AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageServiceBehaviour.swift

```diff
@@ -59,8 +59,7 @@ protocol AWSS3StorageServiceBehaviour {
                          onEvent: @escaping StorageServiceMultiPartUploadEventHandler)
 
     func list(prefix: String,
-              path: String?,
-              onEvent: @escaping StorageServiceListEventHandler)
+              options: StorageListRequest.Options) async throws -> StorageListResult
```

Comment:
Simplifying be removing the whole operation + event handler dance. My understanding was that the main value there as a call to something like:

```
Amplify.Hub.dispatch(to: channel, payload: payload)
```

Which is now done directly in `AWSS3StoragePlugin`